app/src/main/java/com/itsaky/androidide/tasks/git/CloneGitTask.java

```diff
@@ -0,0 +1,195 @@
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import java.util.concurrent.Callable;
+import com.itsaky.androidide.resources.R;
+import com.itsaky.androidide.utils.DialogUtils;
+
+public class CloneGitTask {
+	Activity activity;
+	AlertDialog alertDialog;
+	private static final String TAG = "CloneGitTask";
+	TextInputLayout textInputLayout;
+	TextInputEditText textInputEditText;
+	URL uRL;
+	View view;
+	AlertDialog progressDialog;
+        private Disposable mDisposable;
+        
+	public CloneGitTask(Activity mActivity) {
+		activity = mActivity;
+	}
+
+	public void cloneRepo() {
+		//custom layout
+		LayoutInflater inflater = (LayoutInflater) activity.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+		view = inflater.inflate(R.layout.git_dialog_layout, null);
+
+		textInputLayout = view.findViewById(R.id.textInputRepo);
+		textInputEditText = view.findViewById(R.id.textRepo);
+
+		//materialdialog
+		MaterialAlertDialogBuilder materialAlertDialogBuilder = DialogUtils.newMaterialDialogBuilder(activity);
+		materialAlertDialogBuilder.setView(view);
+		materialAlertDialogBuilder.setTitle(R.string.clone_git_repository);
+		materialAlertDialogBuilder.setCancelable(true);
+		materialAlertDialogBuilder.setPositiveButton(R.string.clone, new DialogInterface.OnClickListener() {
+
+			@Override
+			public void onClick(DialogInterface dia, int which) {
+
+				//url
+				String urlRepo = textInputEditText.getText().toString();
+
+				//materialprogress dialog
+				LayoutInflater inflater = (LayoutInflater) activity.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+				view = inflater.inflate(R.layout.git_dialog_progress, null);
+
+				MaterialAlertDialogBuilder materialAlertDialogBuilder = DialogUtils.newMaterialDialogBuilder(activity);
+				materialAlertDialogBuilder.setTitle(activity.getString(R.string.cloning_repository));
+				materialAlertDialogBuilder.setMessage(urlRepo);
+				materialAlertDialogBuilder.setCancelable(false);
+				materialAlertDialogBuilder.setView(view);
+				
+				materialAlertDialogBuilder.setPositiveButton(R.string.hide, new DialogInterface.OnClickListener() {
+
+					@Override
+					public void onClick(DialogInterface dia, int which) {
+						progressDialog.dismiss();
+					}
+				});
+				materialAlertDialogBuilder.setNegativeButton(R.string.cancel, new DialogInterface.OnClickListener() {
+
+					@Override
+					public void onClick(DialogInterface dia, int which) {
+						mDisposable.dispose();
+					}
+				});
+				progressDialog = materialAlertDialogBuilder.create();
+				progressDialog.show();
+
+				//createObservable
+				createObservable().subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread())
+						.subscribe(new Observer<String>() {
+							@Override
+							public void onSubscribe(Disposable disposable) {
+								Log.d(TAG, "onSubscribe: " + disposable);
```

Comment:
Usage of `android.util.Log` should be replaced with `com.itsaky.androidide.utils.ILogger`.

Reply:
I removed it it says cannot be referenced from a static context